lib/Cake/Controller/Component/AuthComponent.php

```diff
@@ -653,7 +653,7 @@ public function logout() {
  * cookies + sessions will be used.
  *
  * @param string $key field to retrieve. Leave null to get entire User record
- * @return array|null User record. or null if no user is logged in.
+ * @return mixed|null User record. or null if no user is logged in.
```

Comment:
Because expected value is not always array, often it is `string/int/bool` or other types:

I have custom function `myFunction($email)` and $email is defined as `@param string $email`.
IDE highlights `myFunction($this->Auth->user('email'))` as invalid, because it thinks I provide `array|null`, not `string`.

Similar to `Hash:get()`.